From a93e8e1fb8127b9a1ee7567808c9eed2ffe4ccae Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 23 Mar 2018 07:48:13 -0500
Subject: [PATCH 3538/5242] MLK-17703-5: drm: imx: dcss: ignore the 8 bit for
 input pipe

commit  4f642daf8c51e78a8d057e477514757aae566d47 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since the input of HDR10 is always 10-bit, ignore 8-bit flags when
setting up the output pipe.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-hdr10.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dcss/dcss-hdr10.c b/drivers/gpu/imx/dcss/dcss-hdr10.c
index aa4b1e2..7dad29a 100644
--- a/drivers/gpu/imx/dcss/dcss-hdr10.c
+++ b/drivers/gpu/imx/dcss/dcss-hdr10.c
@@ -650,6 +650,6 @@ void dcss_hdr10_setup(struct dcss_soc *dcss, int ch_num,
 	 * Input pipe configuration doesn't matter for configuring the output
 	 * pipe. So, will just mask off the input part of the descriptor.
 	 */
-	dcss_hdr10_pipe_setup(dcss, OPIPE_CH_NO, desc | 0xffff);
+	dcss_hdr10_pipe_setup(dcss, OPIPE_CH_NO, desc | 0xfffe);
 }
 EXPORT_SYMBOL(dcss_hdr10_setup);
-- 
1.7.9.5

